main.tf

```diff
@@ -3,7 +3,7 @@ data "aws_vpc" "vpc" {
 }
 
 resource "aws_elasticache_replication_group" "redis" {
-  replication_group_id          = "${format("%.20s","${var.name}-${var.env}")}"
+  replication_group_id          = "${replace(format("%.20s","${var.name}-${var.env}"), "/\\W+$/", "")}"
```

Comment:
Old good ElasticCache ids... AFAIK this piece is just taking care of forbidden characters: 

> ${replace(format("%.20s", lower(replace("${var.name}-${var.env}", "_", "-"))), "/-+$/", "")}"

Also, a similar solution should be (probably) applied to `aws_elasticache_parameter_group` below.